include/dragonbox/dragonbox.h

```diff
@@ -25,6 +25,32 @@
 #include <limits>
 #include <type_traits>
 
+#ifdef __has_include
+    #if __has_include(<version>)
+        #include <version>
+    #endif
+#endif
+
+#if defined(__cpp_lib_bit_cast) && __cpp_lib_bit_cast >= 201806L
+    #include <bit>
+    #define JKJ_HAS_BIT_CAST 1
+#else
+    #define JKJ_HAS_BIT_CAST 0
+#endif
+
+#if defined(__cpp_lib_is_constant_evaluated) && __cpp_lib_is_constant_evaluated >= 201811L
+    #define JKJ_HAS_IS_CONSTANT_EVALUATED 1
+#else
+    #define JKJ_HAS_IS_CONSTANT_EVALUATED 0
+#endif
+
+// Testing for relevant C++20 constexpr library features
+#if JKJ_HAS_IS_CONSTANT_EVALUATED && JKJ_HAS_BIT_CAST
+    #define JKJ_CONSTEXPR20 constexpr
+#else
+    #define JKJ_CONSTEXPR20
+#endif
+
```

Comment:
Looks all good, but can you make it to use `if consteval` instead of `std::is_constant_evaluated` if it's supported?

Reply:
That's roughly what I thought, yes. But I prefer to have something like

```
#define JKJ_IF_CONSTEVAL if consteval
```

so that I can use it like

```
JKJ_IF_CONSTEVAL {
    /* constexpr code goes here */
}
```

And the reason why I prefer `if consteval` is because it's more capable than `std::is_constant_evaluated`. In the future I might possibly add some things in the `constexpr` path which `std::is_constant_evaluated` cannot really consume. Honestly I personally don't like `std::is_constant_evaluated` at all and consider it a mistake that should be deprecated soon.

Reply:
Should `JKJ_IF_CONSTEVAL` expand to `if (std::is_constant_evaluated())` in C++20 and `if constexpr (false)` in C++17?

I know that `if consteval` is more capable, but here we don't make use of those extra capabilities (mainly being able to call consteval functions within the branch). If the macro gets conditionally expanded to `if (std::is_constant_evaluated())` then calling consteval functions within the branch will fail depending on feature availability. Is it a good idea to overload one macro with all of `if constexpr (false)`, `if(std::is_constant_evaluated())` and `if consteval` depending on feature availability?

To be fair, I agree that `if consteval` is better in every way than `std::is_constant_evaluated()`, except for availability. Having said that I see no reason to deprecate the latter, it's fairly harmless, even if less usable.

Reply:
> hould `JKJ_IF_CONSTEVAL` expand to `if (std::is_constant_evaluated())` in C++20 and `if constexpr (false)` in C++17?

Yes please.

>I know that if consteval is more capable, but here we don't make use of those extra capabilities (mainly being able to call consteval functions within the branch). If the macro gets conditionally expanded to if (std::is_constant_evaluated()) then calling consteval functions within the branch will fail depending on feature availability. Is it a good idea to overload one macro with all of if constexpr (false), if(std::is_constant_evaluated()) and if consteval depending on feature availability?

It sounds weird indeed. But I would say let's just stick to that in this PR. If that causes some problems in the future, then we can deal with that at that point. The intention here is to just consider `if (std::is_constant_evaluated())` as a poor man's replacement for `if consteval`. Someone will find this silly, but in case if this causes troubles, I would introduce another macro for checking if `JKJ_IF_CONSTEVAL` is in fact `if (std::is_constant_evaluated())` in disguise and do branching on that check, rather than always sticking to `if (std::is_constant_evaluated())` except only when `if consteval` is really needed.

Regarding my rant about `std::is_constant_evaluated`, it being co-existing with `if consteval` feels like `std::auto_ptr` being co-existing with `std::unique_ptr`. It's bad pedagogically, a source of subtle bugs, a maintenance burden for implementers/library authors, etc., so I think it's better to deprecate it as soon as possible to prevent it from being more widely used. But obviously that's just my completely subjective opinion.

By the way, I truly appreciate your contribution, thank you a lot!